src/overlays/actors/ovl_Bg_Dblue_Balance/z_bg_dblue_balance.c

```diff
@@ -32,56 +41,689 @@ const ActorInit Bg_Dblue_Balance_InitVars = {
+    s32 pad;
+    WaterBox* sp30;
+    f32 sp2C;
+
+    if (WaterBox_GetSurface1_2(globalCtx, &globalCtx->colCtx, this->dyna.actor.world.pos.x,
+                               this->dyna.actor.world.pos.z, &sp2C, &sp30)) {
+        this->dyna.actor.depthInWater = sp2C - this->dyna.actor.world.pos.y;
+        this->unk_228 = sp2C;
+        if ((this->dyna.actor.depthInWater < 150.0f) && (this->dyna.actor.depthInWater > -150.0f)) {
+            this->unk_182 = true;
+        } else {
+            this->unk_182 = false;
+        }
+    } else {
+        this->unk_182 = false;
+    }
+}
+
+void func_80B8264C(BgDblueBalance* this) {
+    s32 pad;
+    s32 i;
+    BgDblueBalanceStruct* phi_s0;
+    Vec3f sp78;
+    f32 temp_f0;
+    s16 phi_s2;
+
+    if (this->unk_170 != 0) {
+        if (this->unk_183 < 8) {
+            this->unk_183 = 0;
+        } else {
+            this->unk_183 -= 7;
+        }
+    } else {
+        if (this->unk_183 > 250) {
+            this->unk_183 = 255;
+        } else {
+            this->unk_183 += 4;
+        }
+    }
+
+    Matrix_InsertTranslation(this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
+                             MTXMODE_NEW);
+    Matrix_RotateY(this->dyna.actor.shape.rot.y, MTXMODE_APPLY);
+
+    for (i = 0, phi_s0 = &this->unk_188[0], phi_s2 = 0; i < ARRAY_COUNT(this->unk_188);
+         i++, phi_s2 += 0x2000, phi_s0++) {
+        phi_s0->unk_0C = this->dyna.actor.shape.rot.x + phi_s2;
+
+        Matrix_StatePush();
+        Matrix_InsertXRotation_s(phi_s0->unk_0C, MTXMODE_NEW);
+        Matrix_InsertTranslation(-545.0f, 0.0f, 270.0f, MTXMODE_APPLY);
+        Matrix_GetStateTranslation(&sp78);
+        Matrix_StatePop();
+        Matrix_MultiplyVector3fByState(&sp78, &phi_s0->unk_00);
+
+        temp_f0 = (sp78.z - 135.0f);
+        if (sp78.z > 135.0f) {
+            temp_f0 *= 1.8962963f;
+            if (temp_f0 > 255.0f) {
+                phi_s0->unk_0E = 0xFF;
+                phi_s0->unk_10 = 0.1f;
+            } else if (temp_f0 < 1.0f) {
+                phi_s0->unk_0E = 0;
+                phi_s0->unk_10 = 0.0f;
+            } else {
+                phi_s0->unk_0E = temp_f0;
+                phi_s0->unk_10 = temp_f0 * 0.000390625f;
+            }
+        } else {
+            phi_s0->unk_0E = 0;
+        }
+    }
+}
+
+void func_80B828E4(BgDblueBalance* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, BGDBLUEBALANCE_GET_7F(&this->dyna.actor)) && (this->unk_17D <= 0)) {
+        if (this->unk_183 >= 252) {
+            this->unk_183 = 255;
+        } else {
+            this->unk_183 += 3;
+        }
+    } else if (this->unk_183 < 7) {
+        this->unk_183 = 0;
+    } else {
+        this->unk_183 -= 6;
+    }
+}
+
+void func_80B8296C(GlobalContext* globalCtx, Vec3f* arg1, f32 arg2) {
+    s32 i;
+    Vec3f sp60;
+    f32 temp_f0;
+    f32 temp_f20 = arg2 * 2.000000000f;
```

Comment:
Are all those zeroes required?
```suggestion
    f32 temp_f20 = arg2 * 2.0f;
```

Reply:
Yeah they're required.

Reply:
Checkedout the branch to do some testing, and it looks like `arg2 * 2` works.